CIFAR/main.swift

```diff
@@ -27,15 +29,27 @@ for epoch in 1...10 {
         trainingBatchCount += 1
         optimizer.update(&model.allDifferentiableVariables, along: gradients)
     }
-    print("  average loss: \(trainingLossSum / Float(trainingBatchCount))")
-    print("Epoch \(epoch), evaluating on test set...")
+
     var testLossSum: Float = 0
     var testBatchCount = 0
+    var correctGuessCount:Int = 0
+    var totalGuessCount:Int32 = 0
     for batch in testBatches {
         let (labels, images) = (batch.first, batch.second)
         let logits = model.inferring(from: images)
         testLossSum += softmaxCrossEntropy(logits: logits, labels: labels).scalarized()
         testBatchCount += 1
+
+        let correctPredictions = logits.argmax(squeezingAxis: 1) .== labels
+        correctGuessCount = correctGuessCount + Int(Tensor<Int32>(correctPredictions)
+          .sum().scalarized())
+        totalGuessCount = Int32(totalGuessCount) + Int32(batchSize)
```

Comment:
Aren't these already `Int32`?
```suggestion
        totalGuessCount = totalGuessCount + batchSize
```

Reply:
yes, removed